### PR TITLE
Disaster docs moving to get doc build broken

### PR DESCRIPTION
**Pull request by olegtaranenko**
the moving was done to very optimistic level. @carlosmiei Please let me know if you gonna to do similar later

### CODE REVIEW

**Comment by olegtaranenko** [comment]
Currently, I still can't overcome the busted build. It throws the exception at the `creating index of exchange function` stage. Some md's don't generate but the build is finished without exception. It writes in the console names of broken exchanges. I noticed that there are exchanges that have both `src` and `pro` defined classes. No idea why early was OK :-(

**Comment by carlosmiei** [comment]
Hello @olegtaranenko,

Thanks for completing the docs.

Weird, locally I can generate the docs with no issues, which error are you getting?

**Comment by carlosmiei** [comment]
@olegtaranenko

> Please let me know if you gonna to do similar later

What do you mean exactly by this? 😅

**Comment by carlosmiei** [comment]
- relates to https://github.com/ccxt/ccxt/pull/24250

**Comment by olegtaranenko** [comment]
> Actually I'm seeing an error now
> 
> ```
>  node jsdoc2md.js && node examples2md.js
> 
> 📰 loading js docs...
> 📰 rendering docs for each exchange...
> /Users/cjg/Git/ccxt10/ccxt/wiki/helpers.cjs:20
>     return names.map (name => name.replace (/Array./g, 'Array'))
>                  ^
> 
> TypeError: Cannot read properties of undefined (reading 'map')
>     at exports.cleanNames (/Users/cjg/Git/ccxt10/ccxt/wiki/helpers.cjs:20:18)
>     at Object.wrapper (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/internal/wrapHelper.js:15:19)
>     at eval (eval at createFunctionContext (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:262:23), <anonymous>:10:275)
>     at Object.prog [as fn] (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/runtime.js:268:12)
>     at Object.<anonymous> (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/helpers/if.js:29:22)
>     at Object.wrapper (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/internal/wrapHelper.js:15:19)
>     at eval (eval at createFunctionContext (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:262:23), <anonymous>:12:47)
>     at Object.prog [as fn] (/Users/cjg/Git/ccxt10/ccxt/node_modules/handlebars/dist/cjs/handlebars/runtime.js:268:12)
>     at /Users/cjg/Git/ccxt10/ccxt/node_modules/dmd/helpers/helpers.js:116:23
>     at Array.forEach (<anonymous>)
> ```

yes exactly. Please apply my patch and you will get it finished

**Comment by carlosmiei** [comment]
@olegtaranenko But I'm on this branch ` ccxt git:(disaster-docs-moving) ✗ npm run build-docs` and I still get the error 🤔

File: wiki/helpers.cjs [hunk 1]
```diff
@@ -17,6 +17,7 @@ exports.getFragment = function (func) {
 }
 
 exports.cleanNames = function (names) {
+    if (!names) return []
```
Comment (carlosmiei) [inline wiki/helpers.cjs:20]: @olegtaranenko can you try again after this commit? adding this line seems to have fixed it

**Comment by carlosmiei** [comment]
DEMO

```
 npm run build-docs           

> ccxt@4.4.31 build-docs
> node jsdoc2md.js && node examples2md.js

📰 loading js docs...
📰 rendering docs for each exchange...
📰 creating index of exchange functions
📰 finished rendering docs! 🙌 😶‍🌫
📰 Creating Examples.md 
📰 Creating docs for javascript examples
📰 Creating docs for python examples
📰 Creating docs for typescript examples
📰 Creating docs for php examples
```

**Comment by olegtaranenko** [comment]
> @olegtaranenko But I'm on this branch ` ccxt git:(disaster-docs-moving) ✗ npm run build-docs` and I still get the error 🤔



> Hello @olegtaranenko,
> 
> Thanks for completing the docs.
> 
> Weird, locally I can generate the docs with no issues, which error are you getting?

usually before `npm run build-doc` you have to do a full `npm run build`. As I noticed, it takes info from *.js or maybe *.d.ts files

**Comment by carlosmiei** [comment]
@olegtaranenko But i ran `tsc` first to generate the .js file, can you check now with the master branch is the docs are working on your end?

**Comment by olegtaranenko** [comment]
> @olegtaranenko But i ran `tsc` first to generate the .js file, can you check now with the master branch is the docs are working on your end?

It works for me now against the latest master commit. Thanks!

**Pull request merged**